src/main/java/uk/gov/hmcts/darts/audio/deleter/impl/ExternalDataStoreDeleterImpl.java

```diff
@@ -28,32 +28,38 @@ public List<T> delete() {
         List<T> toBeDeleted = finder.findMarkedForDeletion();
 
         for (T entityToBeDeleted : toBeDeleted) {
-            UUID externalLocation = entityToBeDeleted.getLocation();
-            Integer entityId = entityToBeDeleted.getId();
-            int statusId = entityToBeDeleted.getStatusId();
-            log.info(
-                "Deleting storage data with externalLocation={} for entityId={} and statusId={}",
-                externalLocation, entityId, statusId
-            );
-
-            try {
-                deleter.delete(externalLocation);
-            } catch (AzureDeleteBlobException e) {
-                log.error(
-                    "Failed to delete storage data with externalLocation={} for entityId={} and statusId={}",
-                    externalLocation, entityId, statusId, e
-                );
-            }
-
-            repository.delete(entityToBeDeleted);
-            if (entityToBeDeleted instanceof TransientObjectDirectoryEntity transientObjectDirectoryEntity) {
-                log.debug("Deleting transformed media {} with transient object directory id={}",
-                          transientObjectDirectoryEntity.getTransformedMedia().getId(), transientObjectDirectoryEntity.getId());
-                transformedMediaRepository.delete(transientObjectDirectoryEntity.getTransformedMedia());
-            }
-
+            delete(entityToBeDeleted);
         }
         return toBeDeleted;
     }
 
+    @Override
+    public boolean delete(T entityToBeDeleted) {
+        UUID externalLocation = entityToBeDeleted.getLocation();
+        Integer entityId = entityToBeDeleted.getId();
+        boolean deleted = false;
+        int statusId = entityToBeDeleted.getStatusId();
+        log.info(
+            "Deleting storage data with externalLocation={} for entityId={} and statusId={}",
+            externalLocation, entityId, statusId
+        );
+
+        try {
+            deleter.delete(externalLocation);
+            deleted = true;
+        } catch (AzureDeleteBlobException e) {
+            log.error(
+                "Failed to delete storage data with externalLocation={} for entityId={} and statusId={}",
+                externalLocation, entityId, statusId, e
+            );
+        }
+        repository.delete(entityToBeDeleted);
+        if (entityToBeDeleted instanceof TransientObjectDirectoryEntity transientObjectDirectoryEntity
```

Comment:
Be better to override and add this to ExternalOutboundDataStoreDeleter

Reply:
This logic existed previously :)

I just extracted it into a delete method so we can delete singular entries

Reply:
ok just always cautious about instanceof. Theres generally a better place for this logic to sit. This logic does belong in the ExternalOutboundDataStoreDeleter though? It seems the only place where this condition would be true?

Reply:
I would agree it would be better to have this in the more specific super classes so classes that dont needed it dont have it, but that is outside the scope of this ticket as this is existing logic :)